dashboard_api/api/api_v1/endpoints/sites.py

```diff
@@ -13,16 +16,68 @@
     responses={200: dict(description="return a list of all available sites")},
     response_model=Sites,
 )
-def get_sites():
+def get_sites(
+        request: Request,
+        response: Response,
+        cache_client: CacheLayer = Depends(utils.get_cache)):
```

Comment:
endpoint-level caching is borrowed from datasets caching. I think using memcached/elasticache is a bit overkill for the amount of data, but it's already being used, so it's easy to continue that.